Mathlib/Topology/Algebra/Module/Action.lean

```diff
@@ -0,0 +1,238 @@
+However pushforward and products are monotone, so `τ × τ ≤ σ × σ`, and the pushforward of
+`σ × σ` is `≤ σ` because that's precisely the statement that `+` is continuous for `σ`.
+The proof for `•` follows mutatis mutandis.
+
+A *topological module* for a topological ring `R` is an `R`-module `A` with a topology
+making `+` and `•` continuous. The discussion so far has shown that the action topology makes `A`
+into a topological module, and moreover is the finest topology with this property.
+
+A crucial observation is that if `M` is a topological `R`-module, if `A` is an `R`-module with no
+topology, and if `φ : A → M` is linear, then the pullback of `M`'s topology to `A` is a topology
+making `A` into a topological module. Let's for example check that `•` is continuous.
+If `U ⊆ A` is open then by definition of the pullback topology, `U = φ⁻¹(V)` for some open `V ⊆ M`,
+and now the pullback of `U` under `•` is just the pullback along the continuous map
+`id × φ : R × A → R × M` of the preimage of `V` under the continuous map `• : R × M → M`,
+so it's open. The proof for `+` is similar.
+
+As a consequence of this, we see that if `φ : A → M` is a linear map between topological `R`-modules
+modules and if `A` has the action topology, then `φ` is automatically continuous.
+Indeed the argument above shows that if `A → M` is linear then the action
+topology on `A` is `≤` the pullback of the action topology on `M` (because it's the inf of a set
+containing this topology) which is the definition of continuity.
+
+We also deduce that the action topology is a functor from the category of `R`-modules
+(`R` a topological ring) to the category of topological `R`-modules, and it is perhaps
+unsurprising that this is an adjoint to the forgetful functor. Indeed, if `A` is an `R`-module
+and `M` is a topological `R`-module, then the previous paragraph shows that
+the linear maps `A → M` are precisely the continuous linear maps
+from (`A` with its action topology) to `M`, so the action topology is a left adjoint
+to the forgetful functor.
+
+This file develops the theory of the action topology. We prove that the action topology on
+`R` as a module over itself is `R`'s original topology, and that the action topology
+is isomorphism-invariant.
+
+## Main theorems
+
+* `ActionTopology.instSelf : IsActionTopology R R`. The action topology on `R` is `R`'s topology.
+* `ActionTopology.iso [IsActionTopology R A] (e : A ≃L[R] B) : IsActionTopology R B`. If `A` and
+    `B` are `R`-modules with topologies, if `e` is a topological isomorphism between them,
+    and if `A` has the action topology, then `B` has the action topology.
+
+## TODO
+
+Forthcoming PRs from the FLT repo will show that the action topology on a (binary or finite) product
+of modules is the product of the action topologies, and that the action topology on the quotient
+of a module `M` is the quotient topology when `M` is equipped with the action topology.
```

Comment:
I was just about to ask for these, but I guess I don't need to. Leaving them to a future PR is fine!

Reply:
I figured 250 lines was plenty for the PR of the definition (about 50% of them are comments though!)